src/com/codebusters/game/TextParser.java

```diff
@@ -6,72 +6,139 @@
         currentChapter = new Chapter();
         setValidInput(false);
         itemsToAdd = new ArrayList<>();
         itemsToRemove = new ArrayList<>();
+        inventory = new ArrayList<>();
 //         ITEM_VERBS_GAIN = new ArrayList<>();
 //         ITEM_VERBS_GAIN.add("take");
 //         ITEM_VERBS_GAIN.add("grab");
 //         ITEM_VERBS_GAIN.add("pickup");
 //         ITEM_VERBS_GAIN.add("open");
     }
 
+    public static TextParser getInstance(){
+        if (instance == null){
+            instance = new TextParser();
+        }
+        return instance;
+    }
+
     public void parseInput(String input){
-        if (input.equals("")){
-            setValidInput(false);
-        }else{
+        setValidInput(false);   // reset value each time we check
+
+        // check for empty input
+        if (!input.equals("")){
+            // set delims to remove common typing mistakes
             String delims = " \t,.:;?!\"'";
             StringTokenizer parse = new StringTokenizer(input.toLowerCase(), delims);
             ArrayList<String> command = new ArrayList<>();
 
+            // separate input into individual words
             while (parse.hasMoreTokens()){
                 command.add(parse.nextToken());
             }
 
-            if (command.size() != 2){
-                setValidInput(false);
-            }else{
+            // needs to be a two-word input (for now) in the form of verb + noun
+            if (command.size() == 2){
                 String verb = command.get(0);
                 String noun = command.get(1);
 
-                /*
-                @TODO: we have the verb and the noun. We need to check it against list of acceptable ones for the current Chapter. Then take the appropriate action.
-                 */
+                //get current path list
+                ArrayList<HashMap> paths = currentChapter.getPaths();
+                for (HashMap path: paths){
+                    // if we have a valid input
+                    if (verb.equals(path.get("verb")) && noun.equals(path.get("noun"))){
+                        // first, we check for required items
+                        if (!(path.get("requiredItems") == null)){
+//                            String[] reqItems = path.get("requiredItems").split(",");
+                            ArrayList<String> reqItems = (ArrayList<String>) path.get("requiredItems");
+
+                            // we need to make sure at least one required item is in inventory
+                            for (String item: reqItems){
+                                checkAgainstInventory(item.toLowerCase());
+
+                                if (isValidInput()){     // no need to keep searching for others, just need one
+                                    break;
+                                }
+                            }
+
+                            // if a required item is found, then proceed, otherwise invalid input
```

Comment:
Much of this code seems self documenting so the extra code comments are a distraction and unnecessary.